security-analysis/security-analysis-api/src/main/java/com/powsybl/security/limitsreduction/LimitReductionDefinition.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.security.limitsreduction;
+
+import com.google.common.collect.ImmutableList;
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.contingency.ContingencyContext;
+import com.powsybl.iidm.criteria.NetworkElementCriterion;
+import com.powsybl.iidm.criteria.duration.LimitDurationCriterion;
+import com.powsybl.iidm.network.LimitType;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Olivier Perrin {@literal <olivier.perrin at rte-france.com>}
+ */
+public class LimitReductionDefinition {
+    private final LimitType limitType;
+    private final float limitReduction;
```

Comment:
Why a float here?

Reply:
In `LimitViolation`, we already have the notion of limit reduction which is a float. I changed `LimitReductionDefinition.limitReduction` from double to float in order to be consistent.

Reply:
But we plan to remove the detector from the API as it is not used by all the known implementations. It will be in the constructor of the implementation.

Reply:
OK. I will change the type from float to double in a dedicated PR.